cmd/gardener-extension-provider-local/app/app.go

```diff
@@ -331,7 +331,7 @@ func (w *webhookTriggerer) Start(ctx context.Context) error {
 		return err
 	}
 
-	return w.trigger(ctx, w.client, w.client, &appsv1.DeploymentList{}, client.MatchingLabels{"app": "dependency-watchdog-probe"})
+	return w.trigger(ctx, w.client, w.client.Status(), &appsv1.DeploymentList{}, client.MatchingLabels{"app": "dependency-watchdog-probe"})
```

Comment:
Can you explain this? Earlier, we were doing an empty patch with the regular client - why are we doing an empty patch for the status subresource now?

Reply:
Previously, it was:
https://github.com/gardener/gardener/blob/f12b1a3c4ae4acaefac865c4e27d3434482aaf52/cmd/gardener-extension-provider-local/app/app.go#L335
and was having `client.StatusWriter` as parameter, may be this was supposed to be `client.Writer`?

Reply:
For the `Node`s, we want to do a patch with the status subresource. However, for the `Deployment`s, we want to use the regular client.

Reply:
Okay, but the client doesn't implement `StatusWriter` now, so we have to pass it seperately. I will adapt accordingly.